api/src/main/java/org/hawkular/apm/api/model/trace/Node.java

```diff
@@ -213,6 +218,20 @@ public Node setFaultDescription(String faultDescription) {
     }
 
     /**
+     * @return the properties
+     */
+    public Set<Property> getProperties() {
+        return properties;
+    }
+
+    /**
+     * @param properties the properties to set
+     */
+    public void setProperties(Set<Property> properties) {
```

Comment:
Similarly, you might want to create a new Set based on the set you received, to ensure the contents aren't changed by some external object.